pd3/src/main/java/de/dema/pd3/validation/AgeValidator.java

```diff
@@ -22,12 +25,14 @@ public void initialize(Age age) {
     }
 
     @Override
-    public boolean isValid(LocalDate date, ConstraintValidatorContext constraintValidatorContext) {
-        if (date == null) {
-            return false;
-        } else {
-            int age = Period.between(date, LocalDate.now()).getYears();
+    public boolean isValid(String date, ConstraintValidatorContext constraintValidatorContext) {
+        try {
+            LocalDate localDate = LocalDate.parse(date);
+            int age = Period.between(localDate, LocalDate.now()).getYears();
             return age >= minAge && age <= maxAge;
+        } catch (Exception e) {
+            // ignore, format check via pattern...
+            return true;
```

Comment:
Ich glaube nicht, dass ein Validator im Falle einer Exception zurückgeben sollte, dass alles gut ist. Wenn der Validator feststellt, dass kein gültiges Alter vorhanden ist, sollte er false zurückgeben. Abgesehen davon verstehe ich nicht, warum die Annotation immer nur gemeinsam mit der Pattern-Annotation verwendet werden soll und sie sich sogar auf diese verlässt, dass sie den Fehler meldet. Das halte ich schon rein von der Architektur her für unglücklich. Die Annotation sollte in der Lage sein, eigenständig ihre Aufgabe zu erfüllen, ohne von einer anderen Annotation abhängig zu sein.